src/DynamoCoreWpf/Views/Search/LibraryView.xaml.cs

```diff
@@ -146,10 +146,56 @@ private void OnTreeViewItemPreviewMouseLeftButton(object sender, MouseButtonEven
             {
                 if (wrapPanel.MakeOrClearSelection(selectedClass))
                 {
-                    e.Handled = true;
+                    // If class button was clicked, then handle, otherwise leave it.
+                    e.Handled = selectedClass.SubCategories.Count == 0;
                     selectedElement.BringIntoView();
                 }
             }
+
+            ExpandCategory(categoryButton, selectedClass);
+            e.Handled = !(selectedClass is RootNodeCategoryViewModel);
+        }
+
+        private void ExpandCategory(TreeViewItem sender, NodeCategoryViewModel selectedClass)
+        {
+            // Get all category items.
+            var categories = sender.Items.OfType<NodeCategoryViewModel>();
+
+            // Get all current expanded categories.
+            List<NodeCategoryViewModel> allExpandedCategories = categories.
+                Where(cat => (!(cat is ClassesNodeCategoryViewModel) && cat.IsExpanded == true)).ToList();
+
+            var categoryToBeExpanded = categories.Where(cat => cat == selectedClass).FirstOrDefault();
+
+            // If categoryToBeExpanded is null, that means not category button, but class button was clicked.
+            // During loop we will find out to which category this clicked class belongs.
+            if (categoryToBeExpanded != null)
+                categoryToBeExpanded.IsExpanded = !categoryToBeExpanded.IsExpanded;
```

Comment:
No, it can be false as well. Category can be clicked to open and to close. 
![image](https://cloud.githubusercontent.com/assets/8158404/6668539/471079e4-cbfc-11e4-8701-8cbf9fd5cf41.png)
In case above IsExpanded will be false.